### DISCUSSION TITLE
How to use SaveImaged

### DISCUSSION THREAD

**Discussion by JumpLK** [Project-MONAI/MONAI#1678] [Q&A] [answered]
Hi, I'm using the data from the medical segmentation decathlon(the liver ones) : http://medicaldecathlon.com
I'm trying to use SaveImaged to see how my transform affects the data. I tried multiple things :
1) To save images after my transform :
![Capture1](https://user-images.githubusercontent.com/44983802/109682329-9a509480-7b7e-11eb-9792-d93278ae261b.PNG)
It seems like by doing so it tries to allocate a lot of memory (735 Gb):
![Capture1_1](https://user-images.githubusercontent.com/44983802/109682672-eb608880-7b7e-11eb-9896-05978b70a770.PNG)

2) To save images just after the load :
![Capture2](https://user-images.githubusercontent.com/44983802/109682952-28c51600-7b7f-11eb-86a4-37557291eadc.PNG)
It produces an output but it looks corrupted in some way because I'm not able to read it with Slicer3D it makes the software crash

3) To save only the image
![Capture3](https://user-images.githubusercontent.com/44983802/109683892-0da6d600-7b80-11eb-8d40-f4563cc51cb6.PNG)
Same as 2. : It produces an output but it looks corrupted in some way because I'm not able to read it with Slicer3D it makes the software crash

I think that all of thoose errors are because I'm not using SaveImaged as it's intendeed to be used, so do you know how should I use it ?

**Comment by rijobro** [comment, marked as answer]
Hi @JumpLK. `SaveImage` and `SaveImaged` require the image to be passed with the channel dimension first. So firstly you should add `AddChanneld` just after the `LoadImaged`. This also might solve your data allocation problem (currently `Spacingd` and `Orientationd` will be assuming that you have an image that is 512x75 with 512 channels).

I took your code, added the `AddChanneld` and then could open the saved image in ITKSnap. However, image metadata still says that the image is 4D (with a singleton channel dimension). This may not cause any problems for you, but I've submitted a [PR](https://github.com/Project-MONAI/MONAI/pull/1680) to allow the user to squeeze the final dimensions, which would remove the singleton channel dimension, giving a 3D nifti image again.

**Reply by JumpLK** [reply]
Thank you, it works :)
